src/dialogs/edittagdialog.cpp

```diff
@@ -1411,6 +1418,30 @@ void EditTagDialog::FetchTagSongChosen(const Song &original_song, const Song &ne
 
 }
 
+void EditTagDialog::FetchLyrics() {
+
+  if (ui_->song_list->selectionModel()->selectedIndexes().isEmpty()) return;
+  const Song &song = data_[ui_->song_list->selectionModel()->selectedIndexes().first().row()].current_;
+  lyrics_fetcher_->Clear();
+  ui_->lyrics->setPlainText(tr("loading..."));
+  lyrics_id_ = static_cast<qint64>(lyrics_fetcher_->Search(song.effective_albumartist(), song.artist(), song.album(), song.title()));
+
+}
+
+void EditTagDialog::UpdateLyrics(const quint64 id, const QString &provider, const QString &lyrics) {
+
+  Q_UNUSED(provider);
+
+  if (static_cast<qint64>(id) != lyrics_id_) return;
+  lyrics_id_ = -1;
+  if (!lyrics.isEmpty()) {
```

Comment:
```suggestion
  if (lyrics.isEmpty()) {
```
Please move the not found section to the `if`, and setting lyrics to `else`.